nitrocli/src/arg_util.rs

```diff
@@ -107,14 +107,14 @@ macro_rules! enum_int {
     }
 
     impl ::std::str::FromStr for $name {
-      type Err = ();
+      type Err = &'static str;
 
       fn from_str(s: &str) -> ::std::result::Result<Self, Self::Err> {
         match s {
           $(
             $str => Ok($name::$var),
           )*
-          _ => Err(()),
+          _ => Err("[error]"),
```

Comment:
Can this be a meaningful error? Can we include the string and the available variants? Or is it not user-visible? (looks user-visible to me)

You could use something like this, I suppose:
```
    $all
      .iter()
      .map(::std::convert::AsRef::as_ref)
      .collect::<::std::vec::Vec<_>>()
      .join(",")
```

Reply:
Good catch, I just forgot about it.